src/briefcase/integrations/subprocess.py

```diff
@@ -239,13 +248,15 @@ def _run_in_controlled_terminal(self, args, check=False, **kwargs):
         """
         if kwargs.get("stdout") and not kwargs.get("stderr"):
             raise AssertionError(
-                "Subprocess.run() was invoked while dynamic Rich content is active with stdout "
-                "redirected while stderr was not redirected. This would result in stdout "
-                "being printed to the terminal. Redirect stderr as well or use check_output."
+                "Subprocess.run() was invoked while dynamic Rich content is active (or via "
+                "`stream_output`) with stdout redirected while stderr was not redirected. "
+                "This would result in stdout being printed to the console. Redirect stderr "
+                "as well or use check_output."
```

Comment:
Should this say "This would result in *stderr* being printed to the console"?

Either way, I think this sentence is confusing because it isn't really about whether or not something is printed to the console, but whether it prints directly or via streaming. How about changing it to "This would result in output being corrupted or lost"?

Reply:
I ended up adding these assertions because this whole "run with streaming output" has a particularly esoteric operating mode that is almost certainly _not_ what the developer wanted. I've outlined them in this crude table.
```
# | stdout  | stderr  | alert | outcome
--+--------+---------+-------+------------------------------
1 | console | console |       | both stdout and stderr are written to console
2 | pipe    | console |   x   | both stdout and stderr are written to console (except this error is logged instead)
3 | console | pipe    |       | stdout written to console while stderr is piped
4 | pipe    | pipe    |   x   | stdout is written to console while stderr is piped
```

The fourth case is preempted by the fact `Subprocess.run()` will call (the real) `subprocess.run()` and therefore doesn't matter here.

The second case, though, is what triggers this error. This design assumes the interesting output will come from `stdout`; so, it _always_ replaces `stdout` with a pipe so it can be printed in the console.

That's why the error says "This would result in stdout being printed to the console".....because the design assumes that you're trying to get something in the console and that something is in `stdout`.

All that said.....it doesn't bode well for the existing state of this error if it's already being shown to be confusing. Updating to say output would be corrupted or lost isn't really accurate. Such an invocation of `run()` (to result in logging this error) suggests that the dev only wants `stderr` to be written to the console....which just isn't what this code was designed to do. Maybe it makes the most sense to update it to "this is an unsupported configuration for `run()`".

Reply:
Yes, either that or just removing the sentence would be fine. There's no point in spelling out exactly what would happen, since we're not allowing it to happen anyway.